solidity/test/InterchainAccountRouter.t.sol

```diff
@@ -802,9 +802,31 @@ contract InterchainAccountRouterTest is InterchainAccountRouterTestBase {
             routerOverride,
             ismOverride,
             calls,
-            ""
+            bytes("")
         );
         vm.expectCall(address(this), value, data);
         environment.processNextPendingMessage();
     }
+
+    function testSalts() public {
+        address owner = address(this);
+
+        ica = destinationIcaRouter.getDeployedInterchainAccount(
+            origin,
+            owner.addressToBytes32(),
+            address(originIcaRouter).addressToBytes32(),
+            address(environment.isms(destination)),
+            keccak256("i am a salt")
+        );
+
+        OwnableMulticall ica2 = destinationIcaRouter
+            .getDeployedInterchainAccount(
+                origin,
+                owner.addressToBytes32(),
+                address(originIcaRouter).addressToBytes32(),
+                address(environment.isms(destination)),
+                keccak256("i am a different salt")
+            );
+        assertNotEq(address(ica), address(ica2));
+    }
```

Comment:
should we also test the same with the same salt?

Reply:
Done